pkgs/development/libraries/qt-5/5.5/qtbase/setup-hook.sh

```diff
@@ -91,7 +91,7 @@ _qtSetQmakePath() {
 
 if [ -z "$NIX_QT5_TMP" ]; then
     if [ -z "$NIX_QT_SUBMODULE" ]; then
-        NIX_QT5_TMP=$(mktemp -d)
+        NIX_QT5_TMP=$(pwd)/__nix_qt5__
```

Comment:
Also, this will not make the build path less random because Nix still carries out builds in `/tmp`.